environment/integration/helpers.go

```diff
@@ -169,7 +203,7 @@ func initializeDaggerOnce(t *testing.T) {
 	})
 
 	if daggerErr != nil {
-		t.Skipf("Skipping test - Dagger not available: %v", daggerErr)
+		t.Fatalf("Dagger not available: %v", daggerErr)
```

Comment:
🐛 **Bug (Medium):** Changing from Skip to Fatal will cause CI failures when Dagger is unavailable, rather than gracefully skipping tests. This behavioral change may not be intended.

```suggestion
		t.Skipf("Skipping test - Dagger not available: %v", daggerErr)
```